src/vs/workbench/contrib/files/browser/editors/textFileEditor.ts

```diff
@@ -210,18 +213,29 @@ export class TextFileEditor extends BaseTextEditor<ICodeEditorViewState> {
 
 	private openAsBinary(input: FileEditorInput, options: ITextEditorOptions | undefined): void {
 
-		// Mark file input for forced binary opening
-		input.setForceOpenAsBinary();
-
-		// Open in group
-		(this.group ?? this.editorGroupService.activeGroup).openEditor(input, {
+		const defaultBinaryEditor = this.configurationService.getValue<string | undefined>(defaultBinaryEditorSettingId);
+		const groupToOpen = this.group ?? this.editorGroupService.activeGroup;
+		const editorOptions = {
 			...options,
 			// Make sure to not steal away the currently active group
 			// because we are triggering another openEditor() call
 			// and do not control the initial intent that resulted
 			// in us now opening as binary.
 			activation: EditorActivation.PRESERVE
-		});
+		};
+
+		// If we the user setting specifies a default binary editor we use that.
```

Comment:
This block is a bit weird to me, shouldn't we use `replaceEditors` even in the case where no setting is defined? In other words: I would have expected the default setting to configure our default binary editor (which maybe does not exist yet?) and then we could probably even drop the weird `setForceOpenAsBinary` method that I never really liked?

Reply:
Well when undefined we still want to show the binary warning and the Open anyways text which I assume requires you to call open with `setForceOpenAsBinary`

Reply:
This is probably more of a missing adoption for editor resolver in file land to distinguish text vs. binary editor, but I am not even sure how that would work because we only know very very late which editor to use when we attempt to resolve the input: only then do we know that the input is binary, so we cannot make this decision upfront based on the file name really.

Reply:
Well we could have the binary editor registered as a glob that never matches anything and therefore never gets chosen. It will only get chosen if the `id` is passed into the resolver and then the binary editor input would be created. I'm not sure I love it but it would be a way to further centralize things to the resolver.

Reply:
Yeah that sounds like a decent idea, it would allow us to always use `replaceEditor` and not "force open as binary" on the input. We can also take this outside of this PR if it gets too complex.